.github/workflows/macos.yml

```diff
@@ -20,14 +21,21 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [macos-15, macos-14, macos-13]
+        os: [macos-26, macos-15, macos-14]
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v5
       - run: .github/setup-macos.sh
         env:
           KEY_PASSWORD: ${{ secrets.KEY_PASSWORD }}
           PASS_SECRETS_TAR_ENC: ${{ secrets.PASS_SECRETS_TAR_ENC }}
+      - uses: actions/cache@v4
+        id: cache
+        with:
+          path: openssl_bin
+          key: ${{ matrix.os }}-deps-${{ hashFiles('.github/build.sh') }}
```

Comment:
I think we need different cache key here. This just hashes the `.github/build.sh` file as a key, which will not update when we will get a new openssl 3.5 minor update and new commits in that branch.
One option is to force specific openssl version tag, similarly as we do with libressl in linux workflow (but it requires manual updating in some variable with new openssl release).
The other option is to cache just for a week or so like we do here:

https://github.com/latchset/pkcs11-provider/blob/main/.github/workflows/openssl.yml#L40-L51